### DISCUSSION TITLE
Proper way to convert gltf textures to ktx2 for use with bevy

### DISCUSSION THREAD

**Discussion by Severencir** [bevyengine/bevy#11074] [Q&A]
i'm brand new to bevy and clearly doing something wrong, but i have been trying various things and looking up anything i can for days and can't for the life of me figure it out.

i have made a basic shape in blender, added some materials with textures, added a uv map, and exported to glb and gltf. i have tried various methods of attempting to convert it to ktx2 including gltf-transform, gltf-compressor, zencompress, and klafsa. the latter two wont work regardless of what i do, with zencompress just giving me a gltf with no compressed textures, and klafsa giving me errors with the filenames that i can't seem to figure out how to resolve. gltf-transform, and gltf-compressor both seem to work, and give files that show up in the sandbox.babylonjs and gltf-compressor, but both of these when i try to load into bevy give me an error in the following form:

```ERROR bevy_asset::server: Failed to load asset 'MainBlockTest2.glb' with asset loader 'bevy_gltf::loader::GltfLoader': invalid glTF file: missing field `source` at line 1 column 1568```

this is the code i have made for attempting to load the asset:

```use bevy::prelude::*;
use bevy::utils::HashMap;
use std::fs;

#[derive(Resource, Debug, Default)]
pub struct SceneAssets {
    pub scenes: HashMap<String, Handle<Scene>>
}

pub struct AssetLoaderPlugin;

impl Plugin for AssetLoaderPlugin {
    fn build(&self, app: &mut App) {
        app.init_resource::<SceneAssets>().add_systems(Startup, load_assets);
    }
}

fn load_assets(mut scene_assets: ResMut<SceneAssets>, asset_server: Res<AssetServer>){
    let path = "assets";

    let Ok(entries) = fs::read_dir(path) else { return; };

    for entry in entries.filter_map(Result::ok) {
        let path = entry.path();
        if !(path.is_file() && path.extension().unwrap_or_default() == "glb") { continue; }
        
        let Some(filename) = path.file_stem().and_then(|s| s.to_str()) else { continue; };
        
        let asset_path = format!("{}{}", filename, ".glb#Scene0");
        scene_assets.scenes.insert(filename.to_string(), asset_server.load(asset_path));
    }
    
}
```

any help is appreciated

**Comment by TimJentzsch** [comment]
What is the structure of your `asset` folder and what is computed for the `asset_path` in the loop?
That information might help to troubleshoot the issue.

If you plan to work more with Bevy and Blender in combination, I suggest to take a look at @sdfgeoff's [`bevy_blender_toolkit`](https://github.com/sdfgeoff/blender_bevy_toolkit), which appears to make the integration a lot more pleasant.
I haven't used it myself though, so I can't speak from personal experience.

Hope that helps :)
